protos/event.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package farm_ng.core;
+
+enum UriSchemeType {
+  UNKNOWN = 0;
+  FILE = 1;
+  DATA = 2;
+  DEVICE = 3;
+  OTHER = 4;
+}
+
+message Uri {
+  UriSchemeType scheme = 1;
```

Comment:
In general, I prefer string for enum, since it is easier to parse on the c++ side using FARM_ENUM. (Unless size matters, e.g. if we have a tensor of enums...)

But I'm okay to use ints for enums if that makes it easier on the python side - in general.

But for this case, we should use a string for sure since the scheme of a URL is a string by definition ("A non-empty scheme component followed by a colon (:), consisting of a sequence of characters beginning with a letter and followed by any combination of letters, digits").

Reply:
Also a string makes the json representation of the proto human readable and I believe the URI is part of the header and hence should be human readable.